hyperspy/misc/array_tools.py

```diff
@@ -301,6 +308,9 @@ def _linear_bin(dat, scale, crop=True):
         # The new dimension size is old_size/step, this is rounded down normally
         # but if crop is switched off it is rounded up to the nearest whole
         # number.
+        if type(s) != float:
```

Comment:
I don't think this is necessary? Python 3 doesn't do Python 2's "floor division" if the divisor is an int. aka in Python 3, 2/3 = 0.666, not 0.

Reply:
This was necessary to fix a numpy error where the default is to keep same kind.